redbot/core/bank.py

```diff
@@ -229,6 +229,7 @@ async def set_balance(member: Union[discord.Member, discord.User], amount: int)
         ``bank._MAX_BALANCE``.
 
     """
+    amount = int(amount)
```

Comment:
```suggestion
    if not isinstance(amount, int):
        raise TypeError("Amount must be of type int, not {}.".format(type(amount)))
```

Reply:
Can you please also apply this isinstance check to 


Line 847 in the `set_default_balance` function
replace
```py
amount = int(amount)
```
with
```py
if not isinstance(amount, int):
        raise TypeError("Withdrawal amount must be of type int, not {}.".format(type(amount)))
```

Could you please also apply this to `set_max_balance` and `can_spend` 

This will ensure all bank methods only work with integers.